src/fs.h

```diff
@@ -40,6 +43,54 @@ namespace fsbridge {
     };
 
     std::string get_filesystem_error_message(const fs::filesystem_error& e);
+
+// GNU libstdc++ specific workaround for opening UTF-8 paths on Windows.
+//
+// On Windows, it is only possible to reliably access multibyte file paths through
+// `wchar_t` APIs, not `char` APIs. But because the C++ standard doesn't require
+// ifstream/ofstream `wchar_t` constructors, and the GNU library doesn't
+// provide them (in contrast to the Microsoft C++ library, see
+// https://stackoverflow.com/questions/821873/how-to-open-an-stdfstream-ofstream-or-ifstream-with-a-unicode-filename/822032#822032),
+// Boost is forced to fall back to `char` APIs which may not work properly.
```

Comment:
In commit "utils: Add fsbridge fstream function wrapper" (0548f48f19d33cb57b8279d7a647e91d375a916e)

I know this is my own comment, maybe replace "APIs" with "constructors" on this to make clearer this is referring to the same constructors previously mentioned above.